x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.test.tsx

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import '../../../../__mocks__/kea.mock';
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { useValues, useActions } from 'kea';
+
+import { CredentialsList } from './credentials_list';
+import { EuiBasicTable, EuiCopy } from '@elastic/eui';
+import { IApiToken } from '../types';
+import { ApiTokenTypes } from '../constants';
+
+describe('Credentials', () => {
+  const apiToken: IApiToken = {
+    name: '',
+    type: ApiTokenTypes.Private,
+    read: true,
+    write: true,
+    access_all_engines: true,
+    key: 'abc-1234',
+  };
+
+  const mockKea = ({ values = {}, actions = {} }) => {
+    const mergedValues = {
+      apiTokens: [],
+      meta: {
+        page: {
+          current: 1,
+          size: 10,
+          total_pages: 1,
+          total_results: 1,
+        },
+      },
+      ...values,
+    };
+
+    const mergedActions = {
+      deleteApiKey: jest.fn(),
+      fetchCredentials: jest.fn(),
+      showCredentialsForm: jest.fn(),
+      ...actions,
+    };
+
+    (useValues as jest.Mock).mockImplementationOnce(() => mergedValues);
+    (useActions as jest.Mock).mockImplementationOnce(() => mergedActions);
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders', () => {
+    mockKea({});
+    const wrapper = shallow(<CredentialsList />);
+    expect(wrapper.find(EuiBasicTable)).toHaveLength(1);
+  });
+
+  describe('items', () => {
+    it('sorts items by id', () => {
+      mockKea({
+        values: {
+          apiTokens: [
+            {
+              ...apiToken,
+              id: 2,
+            },
+            {
+              ...apiToken,
+              id: undefined,
+            },
+            {
+              ...apiToken,
+              id: 1,
+            },
+          ],
+        },
+      });
+      const wrapper = shallow(<CredentialsList />);
+      const { items } = wrapper.find(EuiBasicTable).props();
+      expect(items.map((i: IApiToken) => i.id)).toEqual([undefined, 1, 2]);
```

Comment:
Ohh! Now I wish I'd read further last night haha, I hadn't realized this awesome sort test was already here. Should we keep both tests still do you think? (I'm fine/don't feel strongly either way, just wanted to get your thoughts)

Reply:
Let's just keep them both.